packages/tracking/src/createTrackingFunction.ts

```diff
@@ -0,0 +1,71 @@
+import { getBaseProperties, type TrackEventProperties } from "./properties.ts";
+
+export interface TrackingOptions {
+    /**
+   * Whether to keep the connection alive for the tracking request.
+   * It is mostly used for tracking links where the user might navigate away before the request is completed.
+   *
+   * Caution! The body size for keepalive requests is limited to 64 kibibytes.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/API/RequestInit#keepalive
+   *
+   * @default false
+   */
+    keepAlive?: boolean;
+}
+
+/**
+ * A function that sends tracking events to the tracking API.
+ * @param eventName The name of the event to track.
+ * @param properties The properties to send with the event.
+ * @param options Options for tracking the event.
+ */
+export type TrackingFunction = (eventName: string, properties: TrackEventProperties, options?: TrackingOptions) => Promise<void>;
+
+// Cannot use URL() because it doesn't support relative base url: https://github.com/whatwg/url/issues/531.
+function resolveApiUrl(path: string, baseUrl: string | undefined) : string {
```

Comment:
One thing I wonder, why a `baseUrl` argument?

Something seems off.

- If the library take care of the tracking endpoint with a default `tracking/track` path, shouldn't it accept an "env" value instead?
- Otherwise, the alternative seems to be to accept a custom `path`. If the consumer knows the path, my guess is that he also know the hostname. Shouldn't it be an absolute url then?

Reply:
I do agree with you with the absolute path making more sense then 2 parts of the url. 

However i know the Performance Management app use their own controller for the telemetry, but the endpoint is /tracking/track also. So i'm not 100% sure if everyone uses the same base.

Also, there is 1 thing that makes me hesitant. I know the  synergy team will want to change /tracking/track to /trackEvent soon-ish. if we let all the product specify the absolute url, i know it will be painful for synergy.

Reply:
Ok ok, I thought there was a "global service" for tracking. If there's no such thing, is there a compelling argument to try to standardize the path to "tracking/track" ?

Reply:
I removed the custom endpoint, this library will provide the tracking/track and let the teams provide their baseUrl. I prefer the baseURL to the env, since it's going to give team more flexibility.